rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinUtilsTest.java

```diff
@@ -1152,6 +1143,57 @@ void addSpendingFederationBaseScript_forSegwitFed_shouldAddExpectedInputWitness(
         assertEquals(oneChunkSegwitScriptSig, transaction.getInput(inputIndex).getScriptSig());
     }
 
+    @Test
+    void buildSegwitScriptSig_buildsScriptSigWithExpectedStructure() {
+        // arrange
+        Federation federation = P2shP2wshErpFederationBuilder.builder().build();
+
+        // act
+        Script segwitScriptSig = buildSegwitScriptSig(federation.getRedeemScript());
+
+        // assert
+        List<ScriptChunk> chunks = segwitScriptSig.getChunks();
+
+        // expected structure is op pushbytes34 + script
+        ScriptChunk chunk = chunks.get(0);
+        int opPushData = chunk.opcode;
+        int expectedOpPushData = 34;
+        assertEquals(expectedOpPushData, opPushData);
+
+        byte[] segwitScriptSigProgram = segwitScriptSig.getProgram();
+        // the whole program sig should start with 22 for op pushbytes34
+        String opPushBytes34 = Hex.toHexString(segwitScriptSigProgram).substring(0, 2);
+        assertEquals("22", opPushBytes34);
+
+        // the script should be 0020[hashed redeem script]
+        // 00 for op 0
+        String op0 = Hex.toHexString(segwitScriptSigProgram).substring(2, 4);
+        assertEquals("00", op0);
+
+        // 20 for op pushbytes32
+        String opPushBytes32 = Hex.toHexString(segwitScriptSigProgram).substring(4, 6);
+        assertEquals("20", opPushBytes32);
+
+        // hashed redeem script
+        byte[] hashedRedeemScript = extractHashedRedeemScriptProgramFromSegwitScriptSig(segwitScriptSig);
+        byte[] expectedHashedRedeemScript = Sha256Hash.hash(federation.getRedeemScript().getProgram());
+        assertArrayEquals(expectedHashedRedeemScript, hashedRedeemScript);
+    }
+
+    @Test
+    void extractHashedRedeemScriptProgramFromSegwitScriptSig_returnsExpectedHash() {
```

Comment:
Missing unit test for NonSegwitScriptSig, Empty, null.

Reply:
Updated.